includes/multi-currency/Locale.php

```diff
@@ -58,32 +57,33 @@ public function get_user_locale_country(): string {
 	 * @return void
 	 */
 	private function load_locale_data() {
-		$locale_info = include WC()->plugin_path() . '/i18n/locale-info.php';
+		$locale_info_path = WC()->plugin_path() . '/i18n/locale-info.php';
 
-		if ( is_array( $locale_info ) && 0 < count( $locale_info ) ) {
-			$countries          = array_keys( $locale_info );
-			$first_country_data = $locale_info[ $countries[0] ];
+		// The full locale data was introduced in the currency-info.php file.
+		// If it doesn't exist we have to use the fallback.
+		if ( ! file_exists( WC()->plugin_path() . '/i18n/currency-info.php' ) ) {
+			$locale_info_path = WCPAY_ABSPATH . 'i18n/locale-info.php';
+		}
 
-			// If the loaded locale_info doesn't contain the locales keys, load the fallback file.
-			if ( is_array( $first_country_data ) && ! array_key_exists( 'locales', $first_country_data ) ) {
-				$locale_info = include WCPAY_ABSPATH . 'i18n/locale-info.php';
-			}
+		$locale_info = include $locale_info_path;
 
+		if ( is_array( $locale_info ) && 0 < count( $locale_info ) ) {
 			// Extract the currency formatting options from the locale info.
-			foreach ( $locale_info as $country => $locale ) {
-				$currency_code = $locale['currency_code'];
+			foreach ( $locale_info as $country_data ) {
```

Comment:
I think we can save some time using: 

```php
$this->currency_format = $locale_info;
```

But it will require changes at places where it's used. What do you think?

Reply:
The main issue would be that `$locale_info` is indexed by country instead of currency, so the lookups would most likely have to iterate to find the currency setting. Is there an easy way around that that I haven't thought of?

Reply:
Nope, you're right, but instead we can load `currency-info.php` file directly if we are going to use only the currency data?

Reply:
I thought about that as well when working on this, but we need the `num_decimals` info, which is present only in the `locale-info` file. Do you think it makes sense to move it to `currency-info` and use it directly here?

Reply:
Every currency has the same decimal numbers for it's locales, right? We can add one line to each currency in that file, and use that directly. You keep this as is right now, and we'll do that in a separate PR. Sounds good?

Reply:
> Every currency has the same decimal numbers for it's locales, right? We can add one line to each currency in that file, and use that directly.

Yep, I think that would work.

> You keep this as is right now, and we'll do that in a separate PR. Sounds good?

All right, sounds good!